File: Abstract Data Types/ObjectLiteral.py
def create_stack():
    items = []

    def push(item):
        items.append(item)

    def pop():
        return items.pop()

    def peek():
        return items[-1]

    def is_empty():
        return len(items) == 0

    def size():
        return len(items)

    return {
        "push": push,
        "pop": pop,
        "peek": peek,
        "is_empty": is_empty,
        "size": size,
    }


stack = create_stack()
stack["push"](1)
stack["push"](2)
stack["push"](3)
print(stack["peek"]())
print(stack["pop"]())
print(stack["size"]())

File: Abstract Data Types/ObjectLiteral.js
function createStack() {
  const items = [];    // private — closure hides it

  return {
    push:    (item) => items.push(item),
    pop:     ()     => items.pop(),
    peek:    ()     => items[items.length - 1],
    isEmpty: ()     => items.length === 0,
    size:    ()     => items.length
  };
}

const stack = createStack();
stack.push(1); stack.push(2); stack.push(3);
console.log(stack.peek()); // 3
console.log(stack.pop());  // 3
console.log(stack.size()); // 2